### PR TITLE
N28: Oxygengine + NavMesh

### PR DESCRIPTION
**Pull request by PsichiX**
_Part of #852_

### CODE REVIEW

**Comment by PsichiX** [comment]
i've used GIF hosted on my repo since these never change and GIF no matter how much i optimize it, it refuses to go down to 2 MB : D

**Comment by PsichiX** [comment]
i've replaced links to external gif with one bound to specific commit to preserve it when file gets moved or replaced or something

File: content/news/028/index.md [hunk 1]
```diff
@@ -336,6 +336,36 @@ Additionally, development has begun on version 0.7 - check out
...
+
+![Oxygengine RPG game template](https://raw.githubusercontent.com/PsichiX/Oxygengine/38c9818069771afcdb7f98eb3c9ee8acf9d4db86/media/oxygengine-overworld.gif)
+
+The hottest HTML5 + WASM game engine for games written in Rust with web-sys.
+
+[@PsichiX], the creator of [Oxygengine], spent last two months on making:
+
+- First few chapters of the book explaining how Oxygengine works in:
+  [Oxygengine Essentials Book][oxygengine-book].
+- New Hardware-Accelerated renderer based on Material Graphs (to allow making
```
Comment (17cupsofcoffee) [inline content/news/028/index.md:349]:
```suggestion
- New hardware-accelerated renderer based on Material Graphs (to allow making
```

File: content/news/028/index.md [hunk 2]
```diff
@@ -336,6 +336,36 @@ Additionally, development has begun on version 0.7 - check out
...
+- New Overworld game module (which aims to provide all essential features needed
+  by RPG game developers, to let them focus on making an actual game).
+- New AI feature module (WIP) that integrates [Emergent AI][emergent] crate with
+  the engine.
+- Plugin-based asset pipeline as well as support for LDtk software projects.
+
+All these changes marks the beggining of stabilizing phase of the engine, API
+more or less won't change much, now the focus is put entirely on the features
+that will push progress towards Ignite visual game editor for artists and game
+designers to use.
```
Comment (17cupsofcoffee) [inline content/news/028/index.md:361]:
```suggestion
All these changes marks the beginning of stabilizing phase of the engine.
The API more or less won't change much - now the focus is put entirely on
the features that will push progress towards the Ignite visual game editor 
for artists and game designers to use.
```

**Review by 17cupsofcoffee — COMMENTED** [review]
Text content looks good, just made a couple of tweaks for readability.

I'm still slightly wary of the size of the image - hosting it elsewhere avoids it taking up space in the repo, but the reader still has to download it (for context, the previous issue of the newsletter was 15mb total, most of which was people's GIFs...). It's not the end of the world, but it's not very friendly to people on mobile/limited data plans.

cc @AngelOnFira, not sure what your thoughts on this are.

File: content/news/028/index.md [hunk 3]
```diff
@@ -336,6 +336,36 @@ Additionally, development has begun on version 0.7 - check out
...
+  [Material Graph based rendering][oxygengine-material-graph] chapter).
+- New Overworld game module (which aims to provide all essential features needed
+  by RPG game developers, to let them focus on making an actual game).
+- New AI feature module (WIP) that integrates [Emergent AI][emergent] crate with
+  the engine.
+- Plugin-based asset pipeline as well as support for LDtk software projects.
+
+All these changes marks the beginning of stabilizing phase of the engine.
+The API more or less won't change much - now the focus is put entirely on
+the features that will push progress towards the Ignite visual game editor
```
Comment (17cupsofcoffee) [inline content/news/028/index.md:360]: whoops
```suggestion
the features that will push progress towards the Ignite visual game editor
```

**Comment by PsichiX** [comment]
if needed i could cut length of the gif till gets 2 mb, but 10 seconds is already short tho

**Comment by AngelOnFira** [comment]
@PsichiX I'm going to make the decision to say that Gifs and other media must be hosted on our repo, and must be smaller than 2.5MB to be included. Although I believe you won't edit that commit, and it should be available, I do still believe that it could cause a headache down the road if something did go wrong. Having the media in our repo does provide a strong guarantee around that.

As for size, that is a strong restriction regardless. Gifs aren't the greatest format for size, maybe we can look into mp4 down the line, but we don't have that now :) Hopefully this works for you!

**Comment by 17cupsofcoffee** [comment]
I'll try to revisit #636 later this month, if I get time - GIF wrangling always ends up being a fairly large chunk of the review process, and I feel like there has to be an easier way :)

**Comment by PsichiX** [comment]
ok, will cut the length (lowering resolution only increases the size xD)

**Comment by PsichiX** [comment]
ok, i've cut length of the gif animation to 2.5 mb @17cupsofcoffee @AngelOnFira

**Review by 17cupsofcoffee — APPROVED** [review]

**Review by AngelOnFira — APPROVED** [review]

**Pull request merged**